rfqmsg/reject.go

```diff
@@ -125,54 +110,42 @@ const (
 	latestRejectVersion = V0
 )
 
-// rejectMsgData is a struct that represents the data field of a quote
-// reject message.
+// rejectMsgData is a struct that represents the data field of a quote reject
+// message.
 type rejectMsgData struct {
 	// Version is the version of the message data.
-	Version WireMsgDataVersion
+	Version tlv.RecordT[tlv.TlvType0, WireMsgDataVersion]
 
 	// ID represents the unique identifier of the quote request message that
 	// this response is associated with.
-	ID ID
+	ID tlv.RecordT[tlv.TlvType2, ID]
 
 	// Err is the error code and message that provides the reason for the
 	// rejection.
-	Err RejectErr
+	Err tlv.RecordT[tlv.TlvType4, RejectErr]
```

Comment:
In the previous TLV setup the err field was optional. I think we want the same here. The peer shouldn't have to provide an error. I think we should use `tlv.TlvType3` here.